packages/jest-types/src/Config.ts

```diff
@@ -27,7 +27,9 @@ export type HasteConfig = {
 export type ReporterConfig = [string, Record<string, unknown>];
 export type TransformerConfig = [string, Record<string, unknown>];
 
-export type ConfigGlobals = Record<string, any>;
+export interface ConfigGlobals {
+  [K: string]: object;
```

Comment:
I think `unknown` is correct, we just need to properly merge the config provided rather than blindly spreading

Reply:
Yeah for sure - I've got a test and everything; just need to come up with a merging solution.

One of the jest packages already has one coded but not exported - I could move that into `jest-utils` (iirc that's a package that exists?).

Otherwise `deepMerge` is used by a few packages, so I'll probably just stick that there 🤷‍♂